client/src/main/scripts/components/search/database/SearchDatabaseApi.ts

```diff
@@ -20,148 +20,83 @@
+    Routes.API_SEARCH,
+    bodyJson
+  );
+
+  const builder = new FoundationDataBuilder(); // This should probably live elsewhere
+  factResults.facts.forEach((fact) => builder.add(fact.hash));
+  const foundationData = await builder.build();
+  return searchImpl(
+    new SearchWithData(searchTerm, factResults.facts, foundationData)
+  );
+}
 
-  static processFacts = (
-    hashesToTurns: HashesToTurns,
-    videoFacts: VideoFactHashMap[]
-  ): FactTurns[] => {
-    let factTurnsArr: FactTurns[] = [];
-    for (const videoFact of videoFacts) {
-      const turns = hashesToTurns.get(videoFact.hash);
-      if (!turns) {
-        throw "Expect hashesToTurns map to have a valid value for each key";
-      }
-      factTurnsArr.push({
-        turns,
-        videoFact: videoFact.videoFact,
-      });
-    }
-    factTurnsArr.sort((aFactTurns, bFactTurns) => {
-      const a = aFactTurns.videoFact.fact.primaryDate;
-      const b = bFactTurns.videoFact.fact.primaryDate;
-      return a == b ? 0 : +(a > b) || -1;
+function searchImpl(searchWithData: SearchWithData): SearchResult {
+  const { foundationData, searchTerm, videoResults } = searchWithData;
+  const hashesToTurns = createHashesToTurns(videoResults);
+  const videoFactsToTurns: VideoFactsToTurns[] = [];
+  for (const [hash, turns] of hashesToTurns) {
+    const videoFact = foundationData.getVideo(hash);
+    turns.sort((a, b) => a - b);
+    videoFactsToTurns.push({
+      turns,
+      videoFact,
     });
-
-    return factTurnsArr;
-  };
+  }
+  videoFactsToTurns.sort((aFactTurns, bFactTurns) => {
+    const a = aFactTurns.videoFact.fact.primaryDate;
+    const b = bFactTurns.videoFact.fact.primaryDate;
+    return a == b ? 0 : +(a > b) || -1;
+  });
+  return new SearchResult(videoFactsToTurns, searchTerm);
 }
 
 export type HashesToTurns = Map<string, number[]>;
 
-export interface FactTurns {
+export interface VideoFactsToTurns {
   videoFact: Foundation.VideoFactContent;
   turns: number[];
 }
 
-export const getSortedFactTurns = (
-  searchResults: Search.FactResultList
-): Promise<FactTurns[]> => {
-  return new Promise((resolve, reject) => {
-    const hashesToTurns = createHashMap(searchResults);
-    if (!hashesToTurns) {
-      return resolve([]);
-    }
-    const innerPromises: Promise<VideoFactHashMap>[] = [];
-    // for (const result of hashesToTurns.keys()) {
-    hashesToTurns.forEach((turns, hash) => {
-      try {
-        innerPromises.push(fetchFactReturningPromise(hash));
-      } catch (err) {
-        reject(err);
-      }
-    });
-
-    Promise.all(innerPromises).then((videoFacts: VideoFactHashMap[]) => {
-      resolve(processFacts(hashesToTurns, videoFacts));
-    });
-  });
-};
-
-export const createHashMap = (
-  results: Search.FactResultList
-): HashesToTurns | undefined => {
-  const facts: Search.VideoResult[] = results.facts;
-  if (facts.length <= 0) {
-    return undefined;
-  }
-
+export const createHashesToTurns = (
```

Comment:
This seems like it should not be an `export`, and instead just a private helper function to `searchImpl`

Reply:
Agreed. The export is a remnant of the previous test harness. This can be fixed now.